### PR TITLE
.github: workflows: Add code_coverage.yml

### LINKED ISSUE #12: Codecoverage workflow
We need a code coverage workflow that collects coverage once for the PHPUnit runner and once for the codeception runner.

Coverage should be collected in a separate workflow and not inside the tests.yaml.

See: https://github.com/php-coveralls/php-coveralls for how to merge several coverage reports into one and sent them to coveralls.io. 

Also, all codeception.dist.yml files need to be configured to collect coverage based on an environment variable.

### PR DESCRIPTION
**Pull request by maluueu**
Closes #12

### CODE REVIEW

File: .github/workflows/code_coverage.yaml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COVERALLS_PARALLEL: true
+          COVERALLS_FLAG_NAME: php-${{ matrix.php }}_wp-${{ matrix.wp }}_dependencies-${{ matrix.dependencies }}_unit
+        run: |
+          composer global require php-coveralls/php-coveralls
+          mkdir build/logs -p
+          mv _output/clover.xml build/logs/
+          php-coveralls -v
+
+#      - uses: actions/upload-artifact@v2
```
Comment (calvinalkan) [inline .github/workflows/code_coverage.yaml:135]: Why is this commented out? @maluueu

File: .github/workflows/code_coverage.yaml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COVERALLS_PARALLEL: true
+          COVERALLS_FLAG_NAME: php-${{ matrix.php }}_dependencies-${{ matrix.dependencies }}_unit
+        run: |
+          composer global require php-coveralls/php-coveralls
+          mkdir build/logs -p
+          mv clover.xml build/logs/
+          php-coveralls -v
+
+#      - uses: actions/upload-artifact@v2
```
Comment (calvinalkan) [inline .github/workflows/code_coverage.yaml:56]: Why is this commented out @maluueu

**Review by calvinalkan — CHANGES_REQUESTED** [review]

**Comment by calvinalkan** [comment]
Can you create a different branch and create a dummy class in any /src folder and create a PR against this branch afterward so we can see how coverage diff looks on a PR?

```php

<?php

class Foo {


public function bar () {
   // uncovered code 
   $foo = 'bar'
}

}

```

File: .github/workflows/code_coverage.yaml (hunk 1 above)
Comment (maluueu) [reply .github/workflows/code_coverage.yaml]: Removed now.

File: .github/workflows/code_coverage.yaml (hunk 2 above)
Comment (maluueu) [reply .github/workflows/code_coverage.yaml]: Removed now.

**Comment by maluueu** [comment]
Change from Coveralls to Codecov

**Comment by calvinalkan** [comment]
Closing this. We'll use codecov.

**Pull request closed without merging**